phac-nml/irida-next
# Data Table: Bot Accounts

## What does this PR do and why?
Note: Due to the size of this current PR, the following will be branched out and fixed in separate PRs:
- Removal of new PAT panel upon bot deletion and PAT revoke

This PR refactors the bot account pages (Group and Project) to use the `data_table_component` (includes PAT tables to use `data_table_component`). This required much of the response logic (turbo etc.) to be re-done. 

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/5902493a-e704-4ea1-a189-e19e1d279cf4)

![image](https://github.com/user-attachments/assets/793cd714-2826-42ed-98e2-0b53b2bdd98f)

## How to set up and validate locally
Test all bot functionality:
1. Create new bot (table updates with new bot and new PAT panel appears/updates)
2. Delete bot (table updates, empty state appears if there are 0 bots)
3. Active PATs for a bot and Revoke action (empty state of PAT table once all PATs are revoked)
4. Generate new PAT (table active PAT # updates, new PAT panel appears/updates)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
